### PR TITLE
Upgrades from Ingress API from extensions/v1beta1 to networking.k8s.io/v1

### PR DESCRIPTION
**Pull request by erickhernandezascencio**
## Checklist
- [x] The title of the PR is clear and informative
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

## Description

Updates k8s version from extensions/v1beta1 to networking.k8s.io/v1. Change is required as previous version will be deprecated in future k8s versions.

## Fixes

N/A

### CODE REVIEW

File: pkg/k8scontext/k8scontext_suite_test.go [hunk 1]
```diff
@@ -13,11 +13,11 @@ import (
 
 	"github.com/onsi/ginkgo"
 	"github.com/onsi/gomega"
-	"k8s.io/klog/v2"
+	// "k8s.io/klog/v2"
```
Comment (akshaysngupta) [inline pkg/k8scontext/k8scontext_suite_test.go:16]: remove comment

File: pkg/tests/fixtures.go [hunk 2]
```diff
@@ -60,72 +60,66 @@ const (
...
-  backend:
-    serviceName: websocket-service
-    servicePort: 80
----
-    `)
-	obj, _, err := scheme.Codecs.UniversalDeserializer().Decode(ingr, nil, nil)
-	if err != nil {
-		return nil, err
+func GetVerySimpleIngress() *networking.Ingress {
+	return &networking.Ingress{
```
Comment (akshaysngupta) [inline pkg/tests/fixtures.go:69]: was this failing to serialize ?

File: pkg/appgw/frontend_listeners.go [hunk 3]
```diff
@@ -222,3 +223,19 @@ func (c *appGwConfigBuilder) groupListenersByListenerIdentifier(cbCtx *ConfigBui
 
 	return listenersByID
 }
+
+func attachFirewallPolicy(cbCtx *ConfigBuilderContext, ingress *networking.Ingress, azConfig *listenerAzConfig) {
```
Comment (akshaysngupta) [inline pkg/appgw/frontend_listeners.go:227]: I think we removed this function.

**Review by akshaysngupta — CHANGES_REQUESTED** [review]
left comments

**Comment by akshaysngupta** [comment]
this pr will need follow up with tests

File: pkg/tests/fixtures.go (hunk 2 above)
Comment (erickhernandezascencio) [reply pkg/tests/fixtures.go:69]: yep, was returning nil

File: docs/how-tos/lets-encrypt.md [hunk 4]
```diff
@@ -121,8 +121,10 @@ Follow the steps below to install [cert-manager](https://docs.cert-manager.io) o
         http:
         paths:
         - backend:
-            serviceName: frontend
-            servicePort: 80
+            service:
+                name: frontend
```
[the remark this answers is not in the document]
Comment (erickhernandezascencio) [reply docs/how-tos/lets-encrypt.md:125]: unsure what you mean. i think indentation is correct?

**Review by akshaysngupta — COMMENTED** [review]
Looking good so far!

File: pkg/appgw/health_probes.go [hunk 5]
```diff
@@ -82,7 +82,7 @@ func (c *appGwConfigBuilder) generateHealthProbe(backendID backendIdentifier) *n
 		return nil
 	}
 	probe := defaultProbe(c.appGwIdentifier, n.HTTP)
-	probe.Name = to.StringPtr(generateProbeName(backendID.Path.Backend.ServiceName, backendID.Path.Backend.ServicePort.String(), backendID.Ingress))
+	probe.Name = to.StringPtr(generateProbeName(backendID.Path.Backend.Service.Name, serviceBackendPortToStr(backendID.Path.Backend.Service.Port), backendID.Ingress))
```
Comment (akshaysngupta) [inline pkg/appgw/health_probes.go:85]: need to make sure there is a test for service == nil

File: pkg/tests/ingress-complex.yaml [hunk 6]
```diff
@@ -15,25 +15,36 @@ spec:
...
         paths:
           - path: /store/*
             backend:
-              serviceName: web-service
-              servicePort: 80
+              service:
+                name: web-service
+                port:
+                  number: 80
+            pathType: Prefix
```
Comment (akshaysngupta) [inline pkg/tests/ingress-complex.yaml:43]: will need to handle pathType

File: pkg/k8scontext/context.go [hunk 7]
```diff
@@ -541,55 +545,95 @@ func (c *Context) GetInfrastructureResourceGroupID() (azure.SubscriptionID, azur
...
-	if err != nil {
-		e := controllererrors.NewErrorWithInnerErrorf(
-			controllererrors.ErrorUpdatingIngressStatus,
-			err,
-			"Unable to get ingress %s/%s", ingressToUpdate.Namespace, ingressToUpdate.Name,
-		)
-		c.MetricStore.IncErrorCount(e.Code)
-		return e
-	}
+func (c *Context) UpdateIngressStatus(ingressToUpdate networking.Ingress, newIP IPAddress) error {
```
Comment (akshaysngupta) [inline pkg/k8scontext/context.go:548]: TODO: check if ingress v1beta1 is updated correctly

**Review by akshaysngupta — APPROVED** [review]
I have added few comments that need to be addressed in subsequent PRs
